manifest.go

```diff
@@ -0,0 +1,655 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package iceberg
+
+import (
+	"io"
+	"sync"
+
+	iceio "github.com/apache/iceberg-go/io"
+
+	"github.com/hamba/avro/v2/ocf"
+)
+
+// ManifestContent indicates the type of data inside of the files
+// described by a manifest. This will indicate whether the data files
+// contain active data or deleted rows.
+type ManifestContent int32
+
+const (
+	ManifestContentData    ManifestContent = 0
+	ManifestContentDeletes ManifestContent = 1
+)
+
+type fieldSummary struct {
+	ContainsNull bool    `avro:"contains_null"`
+	ContainsNaN  *bool   `avro:"contains_nan"`
+	LowerBound   *[]byte `avro:"lower_bound"`
+	UpperBound   *[]byte `avro:"upper_bound"`
+}
+
+type manifestFileV1 struct {
+	Path               string          `avro:"manifest_path"`
+	Len                int64           `avro:"manifest_length"`
+	PartitionSpecID    int32           `avro:"partition_spec_id"`
+	Content            ManifestContent `avro:"content"`
```

Comment:
`content` isn't part of a v1 manifest file. See also https://github.com/apache/iceberg/blob/9907a97351138e676d08e02ea53b746f4e331ec6/core/src/main/java/org/apache/iceberg/V1Metadata.java#L33

Reply:
cc @Fokko

Reply:
it seems to be optional in the Java code as well: https://github.com/apache/iceberg/blob/5e5c6d1849dece373b0d425ebd3ba5c7e98ad0ef/api/src/main/java/org/apache/iceberg/ManifestFile.java#L52C1-L53

Reply:
so it seems that the Spec itself for v1 should have `added_snapshot_id` as `optional`

Reply:
I've opened https://github.com/apache/iceberg/pull/8600 to fix the Spec

Reply:
@zeroshade just FYI that `added_snapshot_id` is mandatory after all for V1 and V2, so you might want to follow-up and make it required

Reply:
sure thing! i'll make a PR for that, thanks for the heads up